app/controllers/spree/admin/taxjar_settings_controller.rb

```diff
@@ -1,17 +1,17 @@
 module Spree
   module Admin
     class TaxjarSettingsController < Spree::Admin::BaseController
+      helper_method :taxjar_tax_category_option
+
       def edit
         @configuration = SuperGood::SolidusTaxjar.configuration
         @nexus_regions = Rails.cache.fetch(:nexus_regions) || []
+        @tax_categories = cache_tax_categories
       end
 
       def update
-        if SuperGood::SolidusTaxjar.configuration.update(configuration_params)
-          flash[:success] = "TaxJar settings updated!"
-        else
-          flash[:alert] = "Failed to update settings!"
-        end
+        configuration_update if params[:super_good_solidus_taxjar_configuration]
+        tax_category_update if params[:tax_category]
```

Comment:
I am not sure how I feel about reusing the same action for both updates here. Is there a reason why we don't want to add a new controller or even a separate action?

This is probably a bigger conversation but we've already started using this controller for various things unrelated to taxjar settings and I would love if we can stop adding more things to it and start splitting the responsibilities into separate controllers.

Reply:
I think from a UX perspective, it is great to have all of these Solidus TaxJar settings on a single page.

I think that having more, non-RESTful controller actions would be better overall. But I think that may be an unpopular opinion.

Reply:
We can still have the settings on a single page and delegate the actions to a different controller though? If we are worried about keeping things RESTful then we've already muddied the waters by not having a separate controller for the resources we're trying to manage here.

Reply:
We're interested in making this change Chris, but maybe in another PR? So that we can go ahead with merging the working functionality.